smartsim/settings/lsfSettings.py

```diff
@@ -0,0 +1,326 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from ..error import SSConfigError
+from ..utils.helpers import init_default
+from .settings import BatchSettings, RunSettings
+
+
+class JsrunSettings(RunSettings):
+    def __init__(self, exe, exe_args=None, run_args=None, env_vars=None):
+        """Settings to run job with ``jsrun`` command
+
+        ``JsrunSettings`` can be used for both the `lsf` launcher.
+
+        :param exe: executable
+        :type exe: str
+        :param exe_args: executable arguments, defaults to None
+        :type exe_args: str | list[str], optional
+        :param run_args: arguments for run command, defaults to None
+        :type run_args: dict[str, str], optional
+        :param env_vars: environment vars to launch job with, defaults to None
+        :type env_vars: dict[str, str], optional
+        """
+        super().__init__(
+            exe, exe_args, run_command="jsrun", run_args=run_args, env_vars=env_vars
+        )
+
+        # Parameters needed for MPMD run
+        self.host = None
+        self.smts_per_task = 4
+        self.mpmd = False
```

Comment:
I think we should implement this as a user space option if we can. Slurm doesn't support this currently, but for Aprun and Mpirun we supply a ``make_mpmd`` method which allows the user to create their own MPMD through combining multiple ``AprunSettings`` or ``MpirunSettings``.


See ``AprunSettings`` for an example